File: VisualBasic/Exercises/StudentManagement/qlhs_lastversion/GUILayer/frmTimKiemHocSinh.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using QuanLyHocSinhPTNK.Services;
using QuanLyHocSinhPTNK.Entities;
using BUSLayer;

namespace Do_An_Cong_Ty
{
    public partial class frmTimKiemHocSinh : Form
    {
        public frmTimKiemHocSinh()
        {
            InitializeComponent();           
        }        
        
        private void btnThoat_Click(object sender, EventArgs e)
        {
            this.Close();
        }

        private void btnTraCuu_Click(object sender, EventArgs e)
        {
            string maHS = " 1=1 ";
            string tenHS = " 1=1 ";
            string ngaysinh = " 1=1 ";
            string diachi = " 1=1 ";
            string toan = " 1=1 ";
            string ly = " 1=1 ";
            string hoa = " 1=1 ";
            string sinh = " 1=1 ";
            string su = " 1=1 ";
            string dia = " 1=1 ";
            string van = " 1=1 ";
            string daoduc = " 1=1 ";
            string theduc = " 1=1 ";
            string DTB = " 1=1 ";
            string lop = " 1=1 ";            
           
            if (maskMaHS.Text != "")
            {
                maHS = "hs.MaHocSinh = " + "'" + maskMaHS.Text + "'" + " ";
            }

            if (txtTenHocSinh.Text != "")
            {
                tenHS = " hs.HoTenHocSinh like " + "'" + txtTenHocSinh.Text + "'" ;
            }

            if (chkNgaySinh.Checked == true)
            {
                ngaysinh = " hs.NgaySinh between " + "'" + dtpNgaySinhTu.Value.ToString() + "'" + " and " + "'" + dtpNgaySinhDen.Value.ToString() + "'" ;
            }

            if (chkDiaChi.Checked == true)
            {
                diachi = " hs.DiaChi like " + "'" + txtDiaChi.Text + "'";
            }

            if (chkDiemToan.Checked == true)
            {
                toan = " d.MaMon = 'MH001' and " + "d.DTB between " + "'" + nudDiemToanTu.Value.ToString() + "'" + " and " + "'" + nudDiemToanDen.Value.ToString() + "'"; 
            }

            if (chkDiemLy.Checked == true)
            {
                ly = " d.MaMon = 'MH002' and " + "d.DTB between " + "'" + nudDiemLyTu.Value.ToString() + "'" + " and " + "'" + nudDiemLyDen.Value.ToString() + "'"; 
            }

            if (chkDiemHoa.Checked == true)
            {
                hoa = " d.MaMon = 'MH003' and " + "d.DTB between " + "'" + nudDiemHoaTu.Value.ToString() + "'" + " and " + "'" + nudDiemHoaDen.Value.ToString() + "'";
            }

            if (chkDiemSinh.Checked == true)
            {
                sinh = " d.MaMon = 'MH004' and " + "d.DTB between " + "'" + nudDiemSinhTu.Value.ToString() + "'" + " and " + "'" + nudDiemSinhDen.Value.ToString() + "'";
            }
            if (chkDiemSu.Checked == true)
            {
                su = " d.MaMon = 'MH005' and " + "d.DTB between " + "'" + nudDiemSuTu.Value.ToString() + "'" + " and " + "'" + nudDiemSuDen.Value.ToString() + "'";
            }

            if (chkDiemDia.Checked == true)
            {
                dia = " d.MaMon = 'MH006' and " + "d.DTB between " + "'" + nudDiemDiaTu.Value.ToString() + "'" + " and " + "'" + nudDiemDiaDen.Value.ToString() + "'";
            }

            if (chkDiemVan.Checked == true)
            {
                van = " d.MaMon = 'MH007' and " + "d.DTB between " + "'" + nudDiemVanTu.Value.ToString() + "'" + " and " + "'" + nudDiemVanDen.Value.ToString() + "'";
            }

            if (chkDaoDuc.Checked == true)
            {
                daoduc = " d.MaMon = 'MH008' and " + "d.DTB between " + "'" + nudDaoDucTu.Value.ToString() + "'" + " and " + "'" + nudDaoDucDen.Value.ToString() + "'";
            }

            if (chkTheDuc.Checked == true)
            {
                theduc = " d.MaMon = 'MH009' and " + "d.DTB between " + "'" + nudTheDucTu.Value.ToString() + "'" + " and " + "'" + nudTheDucDen.Value.ToString() + "'";
            }

            if (chkDTB.Checked == true)
            {
                DTB = " bd.DTB between " + "'" + nudDTBTu.Value.ToString() + "'" + " and " + "'" + nudDTBDen.Value.ToString() +"'";
            }

            if (chkLopHoc.Checked == true)
            {
                lop = " lh.MaLop = " + "'" + cmbLopHoc.SelectedValue.ToString() + "'";
            }          

            DataTable dt = new DataTable();
            HocSinhBUS hsBUS = new HocSinhBUS();
            dt = hsBUS.LayThongTinHocSinh(maHS, tenHS, ngaysinh, diachi, toan, ly, hoa, sinh, su, dia, van, daoduc, theduc, DTB, lop);
            dtgHocSinh.DataSource = dt;
        }

        private void FrmTimKiemHocSinh_Load(object sender, EventArgs e)
        {
            maskMaHS.Mask = "00L0000";

            TList<LopHoc> dslh = new TList<LopHoc>();
            LopHocBUS lhBUS = new LopHocBUS();
            dslh = lhBUS.LayBang();
            cmbLopHoc.DataSource = dslh;
            cmbLopHoc.DisplayMember = "TenLop";
            cmbLopHoc.ValueMember = "MaLop";
        }

        private void chkNgaySinh_CheckedChanged(object sender, EventArgs e)
        {
            if (chkNgaySinh.Checked == true)
            {
                dtpNgaySinhTu.Enabled = true;
                dtpNgaySinhDen.Enabled = true;
            }
            else
            {
                dtpNgaySinhTu.Enabled = false;
                dtpNgaySinhDen.Enabled = false;
            }
        }

        private void chkDiaChi_CheckedChanged(object sender, EventArgs e)
        {
            if (chkDiaChi.Checked == true)
            {
                txtDiaChi.Enabled = true;
            }
            else
            {
                txtDiaChi.Enabled = false;
            }
        }

        private void chkDiemToan_CheckedChanged(object sender, EventArgs e)
        {
            if (chkDiemToan.Checked == true)
            {
                nudDiemToanTu.Enabled = true;
                nudDiemToanDen.Enabled = true;
            }
            else
            {
                nudDiemToanTu.Enabled = false;
                nudDiemToanDen.Enabled = false;
            }
        }

        private void chkDiemLy_CheckedChanged(object sender, EventArgs e)
        {
            if (chkDiemLy.Checked == true)
            {
                nudDiemLyTu.Enabled = true;
                nudDiemLyDen.Enabled = true;
            }
            else
            {
                nudDiemLyTu.Enabled = false;
                nudDiemLyDen.Enabled = false;
            }
        }

        private void chkDiemHoa_CheckedChanged(object sender, EventArgs e)
        {
            if (chkDiemHoa.Checked == true)
            {
                nudDiemHoaTu.Enabled = true;
                nudDiemHoaDen.Enabled = true;
            }
            else
            {
                nudDiemHoaTu.Enabled = false;
                nudDiemHoaDen.Enabled = false;
            }
        }

        private void chkDiemSinh_CheckedChanged(object sender, EventArgs e)
        {
            if (chkDiemSinh.Checked == true)
            {
                nudDiemSinhTu.Enabled = true;
                nudDiemSinhDen.Enabled = true;
            }
            else
            {
                nudDiemSinhTu.Enabled = false;
                nudDiemSinhDen.Enabled = false;
            }
        }

        private void chkDiemSu_CheckedChanged(object sender, EventArgs e)
        {
            if (chkDiemSu.Checked == true)
            {
                nudDiemSuTu.Enabled = true;
                nudDiemSuDen.Enabled = true;
            }
            else
            {
                nudDiemSuTu.Enabled = false;
                nudDiemSuDen.Enabled = false;
            }
        }

        private void chkDiemDia_CheckedChanged(object sender, EventArgs e)
        {
            if (chkDiemDia.Checked == true)
            {
                nudDiemDiaTu.Enabled = true;
                nudDiemDiaDen.Enabled = true;
            }
            else
            {
                nudDiemDiaTu.Enabled = false;
                nudDiemDiaDen.Enabled = false;
            }
        }

        private void chkDiemVan_CheckedChanged(object sender, EventArgs e)
        {
            if (chkDiemVan.Checked == true)
            {
                nudDiemVanTu.Enabled = true;
                nudDiemVanDen.Enabled = true;
            }
            else
            {
                nudDiemVanTu.Enabled = false;
                nudDiemVanDen.Enabled = false;
            }
        }

        private void chkDaoDuc_CheckedChanged(object sender, EventArgs e)
        {
            if (chkDaoDuc.Checked == true)
            {
                nudDaoDucTu.Enabled = true;
                nudDaoDucDen.Enabled = true;
            }
            else
            {
                nudDaoDucTu.Enabled = false;
                nudDaoDucDen.Enabled = false;
            }
        }

        private void chkTheDuc_CheckedChanged(object sender, EventArgs e)
        {
            if (chkTheDuc.Checked == true)
            {
                nudTheDucTu.Enabled = true;
                nudTheDucDen.Enabled = true;
            }
            else
            {
                nudTheDucTu.Enabled = false;
                nudTheDucDen.Enabled = false;
            }
        }

        private void chkDTB_CheckedChanged(object sender, EventArgs e)
        {
            if (chkDTB.Checked == true)
            {
                nudDTBTu.Enabled = true;
                nudDTBDen.Enabled = true;
            }
            else
            {
                nudDTBTu.Enabled = false;
                nudDTBDen.Enabled = false;
            }
        }

        private void chkLopHoc_CheckedChanged(object sender, EventArgs e)
        {
            if (chkLopHoc.Checked == true)
            {
                cmbLopHoc.Enabled = true;
            }
            else
            {
                cmbLopHoc.Enabled = false;
            }
        }

        private void btnThoat_Click_1(object sender, EventArgs e)
        {
            this.Close();
        }       
    }
}

File: VisualBasic/Exercises/SimpleWorkWithDB/3LayersWithPara/ThaoTacCSDLSln/QuanLyHocSinh/frmTimKiemHocSinh.vb
Imports BUSLayer
Imports DTO

Public Class frmTimKiemHocSinh
    Private hsCrt As HocSinhCrt

#Region "Load Data"
    Private Sub LoadLopHoc()
        Dim lhBus As New LopHocBus()
        Dim ds As IList
        ds = lhBus.LayDanhSach()
        cmbLopHoc.DataSource = ds
        cmbLopHoc.ValueMember = "Ma"    'Chuoi ten property Ma cua doi tuong LopHocDto
        cmbLopHoc.DisplayMember = "Ten" 'Chuoi ten property Ten cua doi tuong LopHocDto
    End Sub
    Private Sub CapNhatTieuChuanTraCuu(ByVal bUpdated As Boolean)
        If (bUpdated) Then
            If (txtMaHocSinh.Text <> "") Then
                hsCrt.Ma = Integer.Parse(txtMaHocSinh.Text)
            Else
                hsCrt.Ma = 0
            End If
            hsCrt.Ten = txtTenHocSinh.Text
            hsCrt.NgaySinhTu = dtpNgaySinhTu.Value.Date
            hsCrt.NgaySinhDen = dtpNgaySinhDen.Value.Date
            hsCrt.DiaChi = txtDiaChi.Text
            hsCrt.ToanTu = nudDiemToanTu.Value
            hsCrt.ToanDen = nudDiemToanDen.Value
            hsCrt.LyTu = nudDiemLyTu.Value
            hsCrt.LyDen = nudDiemLyDen.Value
            hsCrt.HoaTu = nudDiemHoaTu.Value
            hsCrt.HoaDen = nudDiemHoaDen.Value
            hsCrt.DTBTu = nudDTBTu.Value
            hsCrt.DTBDen = nudDTBDen.Value
            hsCrt.CheckNgaySinh = chkNgaySinh.Checked
            hsCrt.CheckDiaChi = chkDiaChi.Checked
            hsCrt.CheckToan = chkDiemToan.Checked
            hsCrt.CheckLy = chkDiemLy.Checked
            hsCrt.CheckHoa = chkDiemHoa.Checked
            hsCrt.CheckDTB = chkDTB.Checked
            hsCrt.CheckLopHoc = chkLopHoc.Checked
        Else
            If (hsCrt.Ma <> 0) Then
                txtMaHocSinh.Text = hsCrt.Ma.ToString()
            End If
            txtTenHocSinh.Text = hsCrt.Ten
            dtpNgaySinhTu.Value = hsCrt.NgaySinhTu.Date
            dtpNgaySinhDen.Value = hsCrt.NgaySinhDen.Date
            txtDiaChi.Text = hsCrt.DiaChi
            nudDiemToanTu.Value = hsCrt.ToanTu
            nudDiemToanDen.Value = hsCrt.ToanDen
            nudDiemLyTu.Value = hsCrt.LyTu
            nudDiemLyDen.Value = hsCrt.LyDen
            nudDiemHoaTu.Value = hsCrt.HoaTu
            nudDiemHoaDen.Value = hsCrt.HoaDen
            nudDTBTu.Value = hsCrt.DTBTu
            nudDTBDen.Value = hsCrt.DTBDen
            chkNgaySinh.Checked = hsCrt.CheckNgaySinh
            chkDiaChi.Checked = hsCrt.CheckDiaChi
            chkDiemToan.Checked = hsCrt.CheckToan
            chkDiemLy.Checked = hsCrt.CheckLy
            chkDiemHoa.Checked = hsCrt.CheckHoa
            chkDTB.Checked = hsCrt.CheckDTB
            chkLopHoc.Checked = hsCrt.CheckLopHoc
        End If
    End Sub
    Private Sub frmTimKiemHocSinh_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Try
            LoadLopHoc()
            hsCrt = New HocSinhCrt()
            CapNhatTieuChuanTraCuu(False)
            Dim dtHS As New DataTable()
            dtgHocSinh.DataSource = dtHS
        Catch ex As Exception
            MessageBox.Show(ex.Message)
        End Try
    End Sub
#End Region

#Region "Check Criteria"

    Private Sub chkNgaySinh_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) _
        Handles chkNgaySinh.CheckedChanged
        If (chkNgaySinh.Checked) Then
            dtpNgaySinhTu.Enabled = True
            dtpNgaySinhDen.Enabled = True
        Else
            dtpNgaySinhTu.Enabled = False
            dtpNgaySinhDen.Enabled = False
        End If
    End Sub

    Private Sub chkDiaChi_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) _
        Handles chkDiaChi.CheckedChanged
        If (chkDiaChi.Checked) Then
            txtDiaChi.Enabled = True
        Else
            txtDiaChi.Enabled = False
        End If
    End Sub

    Private Sub chkDiemToan_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) _
        Handles chkDiemToan.CheckedChanged
        If (chkDiemToan.Checked) Then
            nudDiemToanTu.Enabled = True
            nudDiemToanDen.Enabled = True
        Else
            nudDiemToanTu.Enabled = False
            nudDiemToanDen.Enabled = False
        End If
    End Sub

    Private Sub chkDiemLy_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) _
        Handles chkDiemLy.CheckedChanged
        If (chkDiemLy.Checked) Then
            nudDiemLyTu.Enabled = True
            nudDiemLyDen.Enabled = True
        Else
            nudDiemLyTu.Enabled = False
            nudDiemLyDen.Enabled = False
        End If
    End Sub

    Private Sub chkDiemHoa_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) _
        Handles chkDiemHoa.CheckedChanged
        If (chkDiemHoa.Checked) Then
            nudDiemHoaTu.Enabled = True
            nudDiemHoaDen.Enabled = True
        Else
            nudDiemHoaTu.Enabled = False
            nudDiemHoaDen.Enabled = False
        End If
    End Sub

    Private Sub chkDTB_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) _
        Handles chkDTB.CheckedChanged
        If (chkDTB.Checked) Then
            nudDTBTu.Enabled = True
            nudDTBDen.Enabled = True
        Else
            nudDTBTu.Enabled = False
            nudDTBDen.Enabled = False
        End If
    End Sub

    Private Sub chkLopHoc_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) _
        Handles chkLopHoc.CheckedChanged
        If (chkLopHoc.Checked) Then
            cmbLopHoc.Enabled = True
        Else
            cmbLopHoc.Enabled = False
        End If
    End Sub

#End Region

#Region "Xu ly su kien"
    Private Sub btnTraCuu_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnTraCuu.Click
        Try
            CapNhatTieuChuanTraCuu(True)
            Dim hsBus As New HocSinhBus()
            Dim dtHS As DataTable
            dtHS = hsBus.TimKiem(hsCrt)
            dtgHocSinh.DataSource = dtHS
        Catch ex As Exception
            MessageBox.Show(ex.Message)
        End Try
    End Sub

    Private Sub btnThoat_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnThoat.Click
        Me.Close()
    End Sub
#End Region
    
End Class